File: app/src/main/java/jp/java_conf/ikastyle/UI/LoadoutDeleteButton.java
package jp.java_conf.ikastyle.UI;

import android.content.Context;
import android.util.AttributeSet;
import jp.java_conf.ikastyle.Entity.Loadout;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class LoadoutDeleteButton extends FloatingActionButton {
    private Loadout loadout;

    public LoadoutDeleteButton(Context context, AttributeSet attrs){
        super(context, attrs);
    }

    public void setLoadout(Loadout loadout) {
        this.loadout = loadout;
    }

    public Loadout getLoadout() {
        return loadout;
    }
}

File: app/src/main/java/com/splatool/ikastyle/ui/LoadoutDeleteButton.kt
package com.splatool.ikastyle.ui

import android.content.Context
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.splatool.ikastyle.model.data.entity.Loadout
import android.util.AttributeSet
import com.splatool.ikastyle.R
import com.splatool.ikastyle.common.const.GearKind

class LoadoutDeleteButton(context: Context, attrs: AttributeSet?) :
    FloatingActionButton(context, attrs) {
    var loadout: Loadout? = null
    var loadoutId : Int = 0

    init{
        // カスタム属性gearKindの値を取得しフィールド変数gearKindに入れる
        val typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.LoadoutDeleteButton, 0, 0)
        loadoutId = typedArray.getInt(R.styleable.LoadoutDeleteButton_loadoutId, 0)
    }
}